File: glossier/exercise/ruby/lib/exercise/models/location.rb
module Exercise
  class Location
    attr_reader :city, :country, :id, :state

    # city: String
    #   The location's city.
    # country: String
    #   The location's country.
    # id: Integer
    #   The unique id of the location.
    # state: String
    #   The location's state/province.
    def initialize(city:, country:, id:, state:)
      @city, @country, @id, @state = city, country, id, state
    end
  end
end

File: glossier/exercise/javascript/lib/exercise/models/location.js
var exports = module.exports = {}

/**
 * city: String
 *   The location's city.
 * country: String
 *   The location's country.
 * id: Number
 *   The unique id of the location.
 * state: String
 *   The location's state/province.
 */
exports.Location = class Location {
  constructor(city, country, id, state) {
    this.city = city
    this.country = country
    this.id = id
    this.state = state
  }
}
